pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -651,60 +651,60 @@ def multi_head_attention_forward(
     assert bias_k is None
     assert bias_v is None
 
-  q = q.contiguous().view(tgt_len, bsz * num_heads, head_dim).transpose(0, 1)
+  # reshape to (bsz, num_heads, tgt_len, head_dim)
+  q = q.contiguous().view(tgt_len, bsz, num_heads, head_dim).transpose(0, 1).transpose(1, 2)
   if k is not None:
-    k = k.contiguous().view(-1, bsz * num_heads, head_dim).transpose(0, 1)
+    k = k.contiguous().view(-1, bsz, num_heads, head_dim).transpose(0, 1).transpose(1, 2)
   if v is not None:
-    v = v.contiguous().view(-1, bsz * num_heads, head_dim).transpose(0, 1)
+    v = v.contiguous().view(-1, bsz, num_heads, head_dim).transpose(0, 1).transpose(1, 2)
 
   if static_k is not None:
     assert static_k.size(0) == bsz * num_heads
     assert static_k.size(2) == head_dim
-    k = static_k
+    k = static_k.view(bsz, num_heads, -1, head_dim)
```

Comment:
According to the doc: `static_k: (N*num_heads, S, E/num_heads)`.

Did you actually test this? This looks wrong. Shouldn't it be:
```
k = static_k.view(num_heads, bsz, -1, head_dim).transpose(0, 1)
```
?

Reply:
Hm maybe not, and this is fine. I don't really understand from the documentation.

Reply:
No I didn't test this. I'm also not sure but I thought it could be done this way as they have `N` first in the doc